src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -272,26 +282,29 @@ private async ValueTask ProcessChangesAsync(ImmutableSegmentedList<ITextSnapshot
                 Contract.ThrowIfTrue(snapshots.IsDefault || snapshots.IsEmpty);
                 var currentSnapshot = GetLatest(snapshots);
 
-                var classificationService = TryGetClassificationService(currentSnapshot);
-                if (classificationService == null)
+                if (TryGetClassificationService(currentSnapshot) is not (var solutionServices, var classificationService))
                     return;
 
                 var currentDocument = currentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
                 if (currentDocument == null)
                     return;
 
-                var (previousSnapshot, previousDocument, previousRoot) = GetLastProcessedData();
+                SumType<SyntaxNode, Document>? previousDocumentOrRoot = null;
+                if (GetLastProcessedData() is { } data)
+                {
+                    previousDocumentOrRoot = data.lastDocumentOrRoot;
+                }
```

Comment:
Ternary?

Reply:
It's a weird case because I need to lift one element within a tuple to `Nullable<T>`. Is there a specific syntax you were thinking of?

Reply:
`var previousDocumentOrRoot =  GetLastProcessedData() is { lastDocumentOrRoot: var lastDocumentOrRoot }
    ? lastDocumentOrRoot
    : null;`

something like that?

Reply:
➡️ Now simplified